### ISSUE TITLE
[Track One] Tests for Service Fabric Processor error handling are unstable for CI and nightly runs

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#7472]
# Summary

Exception tests for the `ServiceFabricProcessor` have recently begun to be non-deterministic during CI and nightly test runs, failing intermittently but frequently.  

An example failure can be found in the [nightly results fromSeptember 3](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=101902&view=ms.vss-test-web.build-test-results-tab&runId=4070760&resultId=101118&paneView=debug).   The failure looks like:

![image](https://user-images.githubusercontent.com/913445/64260878-a01dd900-cef9-11e9-9770-3da54da94dd2.png)

Another example failure can be found in [PPR #7467](https://github.com/Azure/azure-sdk-for-net/pull/7467/checks?check_run_id=211231449).   The failure looks like:

![image](https://user-images.githubusercontent.com/913445/64261011-d52a2b80-cef9-11e9-8a93-76b0ec71a8e6.png)

# Scope of Work

- Analyze the tests to understand the point of non-determinism and decide on an approach to stabilize for CI and nightly runs.

- Refactor the tests as needed to ensure that the results are deterministic and consistently pass.

# Out of Scope

- Changes or refactoring in other areas.

# Success Criteria

- The suite of tests for the `ServiceFabricProcessor` no longer causes intermittent, but frequent, failures during nightly runs.

- Any changes did not cause regressions elsewhere; CI and nightly test runs pass consistently and reliably with deterministic results.

**Comment by JamesBirdsall** [comment]
Is there any guidance for diagnosing issues which only occur in the testbed? Is it possible to turn on tracing, for example? These tests all run reliably on my dev box, and the only info I currently have from the testbed is which assertion failed, so I'm reduced to guessing, and my best guess (increasing the startup timeout) obviously wasn't correct.

**Comment by jfggdl** [comment]
@jsquire, do you have any insights to James' questions?

**Comment by jsquire** [comment]
I'll ask that @danieljurek and @weshaggard weigh in with thoughts for potentially enabling a more verbose logging or tracing within the pipelines.

**Comment by jsquire** [comment]
For what its worth, I do see these intermittently fail when running locally with the command line runner against the `netcoreapp2.1` framework, both under Linux (WSL) and Windows.   It'll often behave the first time but a few consecutive runs tends to bring out the intermittent failures.  

Specifically, I'm issuing: `dotnet test --framework netcoreapp2.1 --filter TestCategory!=Live`

The main difference in the pipeline is that it is running both the track one and track two tests in parallel across the target platforms.  I'm speculating that the increased amount of resource contention is causing the timing to be more non-deterministic than you're seeing with a focused local run.   I've had to run down a set of these in the track two code as well, and believe that was the underlying cause for those as well.

**Comment by weshaggard** [comment]
>I'll ask that @danieljurek Daniel Jurek FTE and @weshaggard Wes Haggard FTE weigh in with thoughts for potentially enabling a more verbose logging or tracing within the pipelines.

Yes I think it is a good idea to enable some more detailed logs, however I suggest they go to a file instead of the console and we attach the logs as artifacts to the pipeline. @chidozieononiwu is this something you can please file an issue on and add support for?

**Comment by chidozieononiwu** [comment]
> I'll ask that @danieljurek Daniel Jurek FTE and @weshaggard Wes Haggard FTE weigh in with thoughts for potentially enabling a more verbose logging or tracing within the pipelines.
> 
> Yes I think it is a good idea to enable some more detailed logs, however I suggest they go to a file instead of the console and we attach the logs as artifacts to the pipeline. @chidozieononiwu is this something you can please file an issue on and add support for?

New issue filed [here](https://github.com/Azure/azure-sdk/issues/658)

**Comment by JamesBirdsall** [comment]
As far as Service Fabric Processor in particular, I expect to work on a new preview soon, and as part of that I will see if I can reimplement the tests in a more deterministic way.

**Comment by jsquire** [comment]
Adding another test for investigation here:  `Microsoft.Azure.EventHubs.Tests.ServiceFabricProcessor.CheckpointingTests.CheckpointBatchTest`, which has begun misbehaving on .NET Core.    More details can be found in #7492 

I'm going to mark it to be skipped, pending the future test work that James has mentioned.

**Comment by samuelkoppes** [comment]
We are evaluating the state of this issue, and will report back with a status update by 11/4/2020.

**Comment by kasun04** [comment]
The Microsoft.Azure.EventHubs.ServiceFabricProcessor package is in preview and there are no plans for a stable release. This package will soon be deprecated and therefore we won't be taking any changes.
